### PR TITLE
Xrcore refactor

### PR DESCRIPTION
**Pull request by eagleivg**
The result of clearing compiler warnings for xrCore (options -Wall -Wextra -Wshadow -Wformat=2 -Wfloat-equal -Wlogical-op -Wshift-overflow=2 -Wduplicated-cond -Wcast-qual -Wcast-align -D_FORTIFY_SOURCE=2)

### CODE REVIEW

File: src/xrCore/FS.cpp [hunk 1]
```diff
@@ -137,7 +137,7 @@ bool file_handle_internal(pcstr file_name, u32& size, int& file_handle)
 
 void* FileDownload(LPCSTR file_name, const int& file_handle, u32& file_size)
 {
-    void* buffer = xr_malloc(file_size);
+    void* buffer = xr_malloc(file_size + 1); // IDK why need +1 for file size, but without this caused invalid write size 1 for this buffer
```
Comment (drug007) [inline src/xrCore/FS.cpp:140]: It looks like workaround not fix.

**Review by drug007 — APPROVED** [review]

File: src/xrCore/_bitwise.h [hunk 2]
```diff
@@ -1,34 +1,17 @@
...
 IC bool negative(const float f) { return f < 0; }
 IC bool positive(const float f) { return f >= 0; }
 IC void set_negative(float& f) { f = -fabsf(f); }
 IC void set_positive(float& f) { f = fabsf(f); }
-#else
-IC bool negative(const float& f) { return *(unsigned*)&f & fdSGN; }
-IC bool positive(const float& f) { return (*(unsigned*)&f & fdSGN) == 0; }
-IC void set_negative(float& f) { *(unsigned*)&f |= fdSGN; }
-IC void set_positive(float& f) { *(unsigned*)&f &= ~fdSGN; }
-#endif
```
Comment (Xottab-DUTY) [inline src/xrCore/_bitwise.h:31]: This is bad that you removed this code in the `LPCSTR -> pcstr` commit. This should be done in separate commit.

File: src/xrCore/_bitwise.h [hunk 3]
```diff
@@ -1,34 +1,17 @@
...
-#define fdONE 0x03F800000 // 1.0f
-#define fdHALF 0x03F000000 // 0.5f
-#define fdTWO 0x040000000 // 2.0
-#define fdOOB 0x000000000 // "out of bounds" value
-#define fdNAN 0x07fffffff // "Not a number" value
-#define fdMAX 0x07F7FFFFF // FLT_MAX
-#define fdRLE10 0x03ede5bdb // 1/ln10
 
 // integer math on floats
-#ifdef _M_AMD64
```
Comment (Xottab-DUTY) [inline src/xrCore/_bitwise.h:21]: You could replace it with `XR_X64` =)

File: src/xrCore/_bitwise.h (hunk 2 above)
Comment (Xottab-DUTY) [reply src/xrCore/_bitwise.h:31]: Also, bitwise operations may be faster.

File: src/xrCore/xr_vector_defs.h [hunk 4]
```diff
@@ -20,7 +20,7 @@ DEFINE_VECTOR(Fvector*, LPFvectorVec, LPFvectorIt);
 DEFINE_VECTOR(Fcolor, FcolorVec, FcolorIt);
 DEFINE_VECTOR(Fcolor*, LPFcolorVec, LPFcolorIt);
 DEFINE_VECTOR(LPSTR, LPSTRVec, LPSTRIt);
-DEFINE_VECTOR(LPCSTR, LPCSTRVec, LPCSTRIt);
```
Comment (Xottab-DUTY) [inline src/xrCore/xr_vector_defs.h:23]: No, it should be `LPCSTRVec` and `LPCSTRIt` for backwards compatibility.

File: src/xrCore/xr_vector_defs.h [hunk 5]
```diff
@@ -19,7 +19,7 @@ DEFINE_VECTOR(Fvector, FvectorVec, FvectorIt);
 DEFINE_VECTOR(Fvector*, LPFvectorVec, LPFvectorIt);
 DEFINE_VECTOR(Fcolor, FcolorVec, FcolorIt);
 DEFINE_VECTOR(Fcolor*, LPFcolorVec, LPFcolorIt);
-DEFINE_VECTOR(LPSTR, LPSTRVec, LPSTRIt);
+DEFINE_VECTOR(pstr, pstrVec, pstrIt);
```
Comment (Xottab-DUTY) [inline src/xrCore/xr_vector_defs.h]: It should be `LPSTRVec` and `LPSTRIt` for backwards compatibility.

**Review by Xottab-DUTY — CHANGES_REQUESTED** [review]

File: src/xrCore/Animation/Motion.hpp [hunk 6]
```diff
@@ -40,7 +40,7 @@ struct st_BoneMotion
         m_Flags.zero();
         ZeroMemory(envs, sizeof(CEnvelope*) * ctMaxChannel);
     }
-    void SetName(LPCSTR nm) { name = nm; }
+    void SetName(pcstr nm) { name = nm; }
```
Comment (FreeZoneMods) [inline src/xrCore/Animation/Motion.hpp:43]: Why you have 2 commits named 'xrCore: refactor - replace LPCSTR to pcstr ' in the history - 5db1f57 and db10f95?

File: src/xrCore/FS.cpp (hunk 1 above)
Comment (FreeZoneMods) [inline src/xrCore/FS.cpp:140]: It's strange. I guess this behaviour wasn't caused by your previous commits in this PR?  Looks like we should add ifdef for debug mode and force the compiler to print a message.

File: src/xrCore/Animation/Motion.hpp (hunk 6 above)
Comment (eagleivg) [reply src/xrCore/Animation/Motion.hpp:43]: Watch carefully, one LPCSTR, another LPSTR

File: src/xrCore/FS.cpp (hunk 1 above)
Comment (eagleivg) [reply src/xrCore/FS.cpp:140]: This manifested itself long before this PR. Honestly, this place looks weird. Probably, the problem is in the wrong implementation of getting the file size, although I still could not detect differences in file size and buffer size.

**Comment by eagleivg** [comment]
Re-open upon completion.
